packages/adapter-planetscale/src/planetscale.ts

```diff
@@ -97,18 +97,22 @@ class PlanetScaleQueryable<ClientT extends planetScale.Client | planetScale.Tran
 }
 
 function parseErrorMessage(message: string) {
-  const match = message.match(
-    /target: (?:.+?) vttablet: (?<message>.+?) \(errno (?<code>\d+)\) \(sqlstate (?<state>.+?)\)/,
-  )
-
-  if (!match || !match.groups) {
+  const regex = /^(.*) \(errno (\d+)\) \(sqlstate ([A-Z0-9]+)\)/
+  const match = message.match(regex)
+
+  if (match) {
+    const [_, errorMessage, code, sqlstate] = match
+    const originalMessageRegex = /target: .* vttablet: (.*)/
+    const originalMessageMatch = errorMessage.match(originalMessageRegex)
+
+    return {
+      code: Number.parseInt(code, 10),
+      message: originalMessageMatch ? originalMessageMatch[1].trim() : errorMessage,
```

Comment:
Why do we need to match with 2 regexes?

Reply:
This would be to fall back to the previous regex written by @SevInf. I can remove this, and see what happens on the CI, but it felt safer to keep this, so we can fall back to the logic that was already shipped to our users.

Reply:
Update: I'll be merging this PR after the CI finishes (finally the snapshots should be sorted out). I'll then merge the companion engines PR. After, I'll open a tiny PR that gets rid of the "fallback error parsing logic" and observe whether it was really not useful after all.